contracts/BitGuildMarketplace.sol

```diff
@@ -0,0 +1,355 @@
+pragma solidity ^0.4.22;
+
+import "./BitGuildToken.sol";
+import "./BitGuildAccessAdmin.sol";
+import "./BitGuildWhitelist.sol";
+import "./BitGuildFeeOracle.sol";
+import "./SafeMath.sol";
+
+
+// @dev Interface for ERC-721 NFT
+contract ERC721Interface {
+    event Transfer(address indexed _from, address indexed _to, uint256 _tokenId);
+    event Approval(address indexed _owner, address indexed _approved, uint256 _tokenId);
+    event ApprovalForAll(address indexed _owner, address indexed _operator, bool _approved);
+    function balanceOf(address _owner) external view returns (uint256);
+    function ownerOf(uint256 _tokenId) external view returns (address);
+    function safeTransferFrom(address _from, address _to, uint256 _tokenId, bytes data) external;
+    function safeTransferFrom(address _from, address _to, uint256 _tokenId) external;
+    function transferFrom(address _from, address _to, uint256 _tokenId) external;
+    function approve(address _approved, uint256 _tokenId) external;
+    function setApprovalForAll(address _operator, bool _approved) external;
+    function getApproved(uint256 _tokenId) external view returns (address);
+    function isApprovedForAll(address _owner, address _operator) external view returns (bool);
+}
+
+/// @title ERC-721 Non-Fungible Token Standard
+interface ERC721TokenReceiver {
+	function onERC721Received(address _from, uint256 _tokenId, bytes data) external returns(bytes4);
+	function onERC721Received(address _operator, address _from, uint256 _tokenId, bytes data) external returns(bytes4);
+}
+
+/*
+ * @title BitGuildMarketplace
+ * @dev: Marketplace smart contract for BitGuild.com
+ */
+contract BitGuildMarketplace is BitGuildAccessAdmin, BitGuildWhitelist {
```

Comment:
You definitely do not want to inherit it from the whitelist! You want to have a changeable whitelist link!